### ISSUE TITLE
Unclosed client session warning is getting displayed at the end of command execution for sync.

### ISSUE THREAD

**Issue by khusbu-crest** [elastic/connectors-python#569]
## Bug Description
An unclosed client session warning is getting displayed at the end of command execution for sync.

### To Reproduce
1. Check out the latest code and Go to the directory.
2. Create the Index and change the required configurations for the connector, in the UI, to ingest the data into the elastic search.
3. Execute the command - make run and perform the sync.
4. Wait till the execution gets completed and check if the attached warning is getting displayed after the execution is completed.

## Expected behavior
There should not be any warning displayed and the execution should be completed successfully without any warning.

## Screenshots

![unclosed client session warning is shown after execution1](https://user-images.githubusercontent.com/96513296/222679717-997c4dc4-4a52-4a99-acaa-7a9960abe16e.png)

## Environment

- OS: Alma Linux8
- Browser - Chrome
- Version 110.0.5481.178

## Additional context
This is happening while executing any connector, every time, after taking the latest pull this week.
Also, if you can confirm if this is the expected behavior or not?

**Comment by tarekziade** [comment]
Thanks! yeah this is definitely an issue about some tasks not being properly cleaned

**Comment by tarekziade** [comment]
it looks like we don't have this problem in `e2e` nightlies, https://buildkite.com/elastic/connectors-python-nightly/builds/389#0186c59d-57e3-4b5f-9e39-cb1b0f1c846e

> Wait till the execution gets completed and check if the attached warning is getting displayed after the execution is completed

do you force-stop the service to get to that point?

**Comment by khusbu-crest** [comment]
@tarekziade Yeah, it is working fine for e2e but while executing make run, this warning appears at the end, without doing any force stop.
